packages/shared_preferences/shared_preferences_platform_interface/lib/shared_preferences_platform_interface.dart

```diff
@@ -0,0 +1,80 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+abstract class SharedPreferencesPlatform {
+  /// Returns all keys in the persistent storage.
+  Set<String> getKeys();
+
+  /// Reads a value of any type from persistent storage.
+  dynamic get(String key);
```

Comment:
`_getSharedPreferencesMap` above returns a (Future) `Map<String, Object>`. Should this getter return `Object` instead of `dynamic`?

Reply:
I think That would be a breaking change. Today you can do: `preferences.get("some_string").substring(0, 3)`. If I change it to `Object` you'd get a compiler error complaining that `Object` doesn't have the `substring` method.

Reply:
Ouchhhh